Integration tests for xtend-archetype failing on jenkins

Tests are passing on my and @spoenemann machine and failing on jenkins.
I've disabled them in release branch.

@LorenzoBettini could you have a look?



```
[INFO] --- maven-archetype-plugin:2.3:integration-test (default-integration-test) @ xtend-archetype ---
[INFO] Processing Archetype IT project: first
[INFO] ----------------------------------------------------------------------------
[INFO] Using following parameters for creating project from Archetype: xtend-archetype:2.11.0
[INFO] ----------------------------------------------------------------------------
[INFO] Parameter: groupId, Value: integrationtest.group
[INFO] Parameter: artifactId, Value: integrationtest.artifactId
[INFO] Parameter: version, Value: 1.0.0-SNAPSHOT
[INFO] Parameter: package, Value: org.eclipse.xtend.xtend.archetype.integrationtest
[INFO] Parameter: packageInPathFormat, Value: org/eclipse/xtend/xtend/archetype/integrationtest
[INFO] Parameter: packageInPathFormat, Value: org/eclipse/xtend/xtend-archetype/integrationtest
[INFO] Parameter: package, Value: org.eclipse.xtend.xtend.archetype.integrationtest
[INFO] Parameter: version, Value: 1.0.0-SNAPSHOT
[INFO] Parameter: sourceEncoding, Value: UTF-8
[INFO] Parameter: groupId, Value: integrationtest.group
[INFO] Parameter: artifactId, Value: integrationtest.artifactId
[ERROR] ResourceManager : unable to find resource 'archetype-resources/pom.xml' in any resource loader.
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building xtend-android-archetype 2.11.0
[INFO] -----------------------------------------
```

--- a/full-build.sh
+++ b/full-build.sh
@@ -2,7 +2,6 @@
 
 MVN_ARGS=(\
   -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
-  -Dit-archetype-tests-skip=true \
 )
 
 # args parsing inspired by https://gist.github.com/jehiah/855086
@@ -22,7 +21,7 @@ while [ "$1" != "" ]; do
       exit
       ;;
     --no-tests)
-      MVN_ARGS+=(-DskipTests=true)
+      MVN_ARGS+=(-DskipTests=true -Dit-archetype-tests-skip=true)
       ;;
     --local-repository)
       MVN_ARGS+=" -Dmaven.repo.local=$VALUE"

--- a/org.eclipse.xtend.maven.archetype/README.md
+++ b/org.eclipse.xtend.maven.archetype/README.md
@@ -1,0 +1,5 @@
+To run the integration tests, the `maven.home` variable must defined, i.e., passed to the Maven build with `-Dmaven.home=...`.
+
+For example, if the Maven executable is `/usr/local/bin/mvn`, you need to pass `-Dmaven.home=/usr/local` (i.e., without the `bin` subdirectory).
+
+The Maven Invoker Plugin's goal `install` is attached to the `pre-integration-test` phase, in order to install the needed Maven artifacts of Xtext/Xtend into the `target/.m2test` folder, which will be then used by the integration test to build the test project generated from the archetype.

--- a/org.eclipse.xtend.maven.archetype/pom.xml
+++ b/org.eclipse.xtend.maven.archetype/pom.xml
@@ -12,7 +12,8 @@
 
 	<properties>
 		<root-dir>${basedir}/..</root-dir>
-		<it-archetype-tests-skip>true</it-archetype-tests-skip>
+		<it-archetype-tests-skip>false</it-archetype-tests-skip>
+		<it-archetype-tests-debug>false</it-archetype-tests-debug>
 	</properties>
 
 	<build>
@@ -53,14 +54,16 @@
 				<artifactId>maven-archetype-plugin</artifactId>
 				<configuration>
 					<skip>${it-archetype-tests-skip}</skip>
-					<debug>true</debug>
+					<debug>${it-archetype-tests-debug}</debug>
 				</configuration>
 				<executions>
 					<execution>
 						<goals>
 							<goal>integration-test</goal>
 						</goals>
-						<configuration><debug>true</debug></configuration>
+						<configuration>
+							<debug>${it-archetype-tests-debug}</debug>
+						</configuration>
 					</execution>
 				</executions>
 			</plugin>
@@ -84,7 +87,7 @@
 				</file>
 			</activation>
 			<properties>
-				<it-tests-skip>true</it-tests-skip>
+				<it-archetype-tests-skip>true</it-archetype-tests-skip>
 			</properties>
 		</profile>
 	</profiles>

--- a/org.eclipse.xtend.maven.archetype/pom.xml
+++ b/org.eclipse.xtend.maven.archetype/pom.xml
@@ -14,6 +14,8 @@
 		<root-dir>${basedir}/..</root-dir>
 		<it-archetype-tests-skip>false</it-archetype-tests-skip>
 		<it-archetype-tests-debug>false</it-archetype-tests-debug>
+		<!-- The local Maven repository for our IT tests -->
+		<localMavenRepositoryPath>${project.basedir}/target/.m2test</localMavenRepositoryPath>
 	</properties>
 
 	<build>
@@ -50,11 +52,37 @@
 			</plugins>
 		</pluginManagement>
 		<plugins>
+			<!-- install test dependencies, including the xtend-maven-plugin in the local .m2 repository
+				that will be used by the integration test to build the test project generated with the archetype -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-invoker-plugin</artifactId>
+				<configuration>
+					<extraArtifacts>
+						<!-- our BOM is not a dependency (it's just imported), thus, we must
+							install it explicitly in the local maven repository -->
+						<extraArtifact>org.eclipse.xtext:xtext-dev-bom:${project.version}:pom</extraArtifact>
+					</extraArtifacts>
+					<localRepositoryPath>${localMavenRepositoryPath}</localRepositoryPath>
+				</configuration>
+				<executions>
+					<execution>
+						<id>install dependencies in local test repo</id>
+						<phase>pre-integration-test</phase>
+						<goals>
+							<goal>install</goal>
+						</goals>
+						<configuration>
+							<scope>test</scope>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
 			<plugin>
 				<artifactId>maven-archetype-plugin</artifactId>
 				<configuration>
-					<skip>${it-archetype-tests-skip}</skip>
 					<debug>${it-archetype-tests-debug}</debug>
+					<localRepositoryPath>${localMavenRepositoryPath}</localRepositoryPath>
 				</configuration>
 				<executions>
 					<execution>
@@ -62,7 +90,7 @@
 							<goal>integration-test</goal>
 						</goals>
 						<configuration>
-							<debug>${it-archetype-tests-debug}</debug>
+							<skip>${it-archetype-tests-skip}</skip>
 						</configuration>
 					</execution>
 				</executions>

--- a/org.eclipse.xtend.maven.archetype/src/main/resources/META-INF/maven/archetype-metadata.xml
+++ b/org.eclipse.xtend.maven.archetype/src/main/resources/META-INF/maven/archetype-metadata.xml
@@ -9,7 +9,7 @@
 			<defaultValue>UTF-8</defaultValue>
 		</requiredProperty>
 		<requiredProperty key="javaVersion">
-			<defaultValue>1.8</defaultValue>
+			<defaultValue>17</defaultValue>
 		</requiredProperty>
 	</requiredProperties>
 	<fileSets>

--- a/org.eclipse.xtend.maven.archetype/src/test/resources/projects/first/archetype.properties
+++ b/org.eclipse.xtend.maven.archetype/src/test/resources/projects/first/archetype.properties
@@ -2,5 +2,6 @@ sourceEncoding=UTF-8
 groupId=integrationtest.group
 artifactId=integrationtest.artifactId
 version=1.0.0-SNAPSHOT
+javaVersion=17
 package=org.eclipse.xtend.xtend.archetype.integrationtest
 packageInPathFormat=org/eclipse/xtend/xtend-archetype/integrationtest

--- a/org.eclipse.xtend.maven.parent/pom.xml
+++ b/org.eclipse.xtend.maven.parent/pom.xml
@@ -110,6 +110,10 @@
 					<artifactId>maven-source-plugin</artifactId>
 					<version>3.2.1</version>
 				</plugin>
+				<plugin>
+					<artifactId>maven-invoker-plugin</artifactId>
+					<version>3.5.0</version>
+				</plugin>
 			</plugins>
 		</pluginManagement>
 	</build>

--- a/org.eclipse.xtend.maven.plugin/pom.xml
+++ b/org.eclipse.xtend.maven.plugin/pom.xml
@@ -174,7 +174,6 @@
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-invoker-plugin</artifactId>
-				<version>3.5.0</version>
 				<configuration>
 					<extraArtifacts>
 						<!-- our BOM is not a dependency (it's just imported), thus, we must

